src/server/scripts/Northrend/VioletHold/boss_zuramat.cpp

```diff
@@ -53,26 +56,29 @@ class boss_zuramat : public CreatureScript
 
     struct boss_zuramatAI : public ScriptedAI
     {
-        boss_zuramatAI(Creature* creature) : ScriptedAI(creature)
+        boss_zuramatAI(Creature* creature) : ScriptedAI(creature), sentries(me)
         {
-            Initialize();
             instance = creature->GetInstanceScript();
         }
 
         void Initialize()
         {
-            SpellShroudOfDarknessTimer = 22000;
-            SpellVoidShiftTimer = 15000;
-            SpellSummonVoidTimer = 12000;
+            events.Reset();
             voidDance = true;
         }
 
-        InstanceScript* instance;
-
-        uint32 SpellVoidShiftTimer;
-        uint32 SpellSummonVoidTimer;
-        uint32 SpellShroudOfDarknessTimer;
-        bool voidDance;
+        void DespawnSentries()
+        {
+            sentries.DespawnAll();
+            std::list<Creature*> sentries;
```

Comment:
This grid search is redundant - the sentry balls are summoned by him thus already added to "sentries" at JustSummoned()